File: UI/wwwroot/Model/Materias.js
//@ts-check
// @ts-ignore
import { EntityClass } from '../WDevCore/WModules/EntityClass.js';
import { Asignaturas } from './Asignaturas.js'
import { Clases } from './Clases.js'
import { Docente_materias } from './Docente_materias.js'
import { Evaluaciones } from './Evaluaciones.js'
class Materias extends EntityClass {

    /** @param {Partial<Materias>} [props] */
    constructor(props) {
        super(props, 'GestionCursos');
        for (const prop in props) {
            this[prop] = props[prop];
        };
    }
    /**@type {Number}*/ Id;
    /**@type {String}*/ Observaciones;
    /**@type {String}*/ Descripcion;
    /**@type {String}*/ Config;
    /**@type {Date}*/ Created_at;
    /**@type {Date}*/ Updated_at;
    /**@type {Number}*/ Lock_version;
    /**@type {Number}*/ Clase_id;
    /**@type {Asignaturas} ManyToOne*/ Asignaturas;
    /**@type {Clases} ManyToOne*/ Clases;
    /**@type {Array<Docente_materias>} OneToMany*/ Docentes_materias;
    /**@type {Array<Evaluaciones>} OneToMany*/ Evaluaciones;
    /**
    * @returns {Promise<Array<Materias>>}
    */
    async GetOw() {
        return await this.GetData("ApiGestionCursos/GetOwMaterias");
    }
}
export { Materias }

File: AppGenerateFiles/dbo/Model/Materias.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Materias : EntityClass {
       [PrimaryKey(Identity = false)]
       public int? Id { get; set; }
       public int? Clase_id { get; set; }
       public int? Asignatura_id { get; set; }
       public string? Observaciones { get; set; }
       public string? Config { get; set; }
       public DateTime? Created_at { get; set; }
       public DateTime? Updated_at { get; set; }
       public int? Lock_version { get; set; }
       [ManyToOne(TableName = "Asignaturas", KeyColumn = "Id", ForeignKeyColumn = "Asignatura_id")]
       public Asignaturas? Asignaturas { get; set; }
       [ManyToOne(TableName = "Clases", KeyColumn = "Id", ForeignKeyColumn = "Clase_id")]
       public Clases? Clases { get; set; }
       [OneToMany(TableName = "Docente_materias", KeyColumn = "Id", ForeignKeyColumn = "Materia_id")]
       public List<Docente_materias>? Docente_materias { get; set; }
       [OneToMany(TableName = "Evaluaciones", KeyColumn = "Id", ForeignKeyColumn = "Materia_id")]
       public List<Evaluaciones>? Evaluaciones { get; set; }
   }
}
